examples/vl3_basic/vl3-nse/cmd/nse.go

```diff
@@ -127,44 +130,32 @@ func main() {
 	mainFlags := &Flags{}
 	mainFlags.Process()
 
-	//var defCEAddon defaultCompositeEndpointAddon
-	ucnfNse := ucnf.NewUcnfNse(mainFlags.ConfigPath, mainFlags.Verify, &vppagent.UniversalCNFVPPAgentBackend{}, CompositeEndpointPlugin)
-
-	defer ucnfNse.Cleanup()
-	<-c
-}
+	cnfConfig := &nseconfig.Config{}
+	f, err := os.Open(mainFlags.ConfigPath)
+	if err != nil {
+		logrus.Fatal(err)
+	}
+	err = nseconfig.NewConfig(yaml.NewDecoder(f), cnfConfig)
+	if err != nil {
+		logrus.Fatal(err)
+	}
 
-/*
-var (
-	nsmEndpoint *endpoint.NsmEndpoint
-)
+	configuration := common.FromEnv()
 
-func main() {
+	backend := &vppagent.UniversalCNFVPPAgentBackend{}
+	if err := backend.NewUniversalCNFBackend(); err != nil {
+		logrus.Fatal(err)
+	}
 
-	// Capture signals to cleanup before exiting
-	c := tools.NewOSSignalChannel()
+	pe := config.NewProcessEndpoints(backend, cnfConfig.Endpoints, configuration, CompositeEndpointPlugin)
```

Comment:
did you have a need to remove the common UcnfNse layer?  The intent of that was to make it so a specific NSE instance could only add its additional composite endpoints to the common UCNF composite endpoints and make this `NewProcessEndpoints` and `pe.Process` done in the common UCNF code.  The idea being that a new type of vppagent NSE would be simpler to implement.

Reply:
I removed It because the new config doesn't have InitActions and without the init actions there was so little code in the `UcnfNse` implementation that it seemed like a useless layer to me
as you said it just wraps `config.NewProcessEndpoints` and `pe.Process` in a new object ...

Reply:
Added back UcnfNse for compatibility with the old code and for less confusion